content/en/docs/concepts/architecture/garbage-collection.md

```diff
@@ -166,17 +166,25 @@ is restarted, the tracked image age is reset, causing the kubelet to wait the fu
 based on image age.
 {{< /note>}}
 
-### Container garbage collection {#container-image-garbage-collection}
+### Container garbage collection {#container-image-garbage-collection} (deprecated)
+
+{{< caution >}}
+Container garbage collection is deprecated in favour of `--eviction-hard` or
+`--eviction-soft` and will be removed in a future version. For more details on
```

Comment:
container gc and eviction are different things ...

Reply:
Hello @tengqm,

> container gc and eviction are different things ...

Can you clarify on the below points of container gc?
1. Is `--minimum-container-ttl-duration`, `--maximum-dead-containers` and `--maximum-dead-containers-per-container` related to container gc?
2. In which version container gc got deprecated?
3.  Is there any alternate recommended way for container garbage collection? What it is?

Reply:
Container GC is a task performed by kubelet regularly. The command line flags you mentioned are about container GC. The things that got deprecated, IIRC, are the command line flags. The intent of the deprecation was to reduce knobs for users to tune.
With the mechanism still there in kubelet, container GC won't go away any time soon. I'm not aware of anyone pushing this forward. In other words, the "deprecation" itself may get rotten (or deprecated).

Reply:
Thanks for sharing feedback @tengqm.


What others say on this? Please share your opinion.

Reply:
> Container GC is a task performed by kubelet regularly. The command line flags you mentioned are about container GC. The things that got deprecated, IIRC, are the command line flags.

That looks right. Most kubelet command line arguments are deprecated (you should use the configuration file), and also the settings for container / container image GC were a bit legacy anyway.